packages/react/src/components/UIShell/HeaderContainer.js

```diff
@@ -14,6 +14,14 @@ const HeaderContainer = ({ isSideNavExpanded, render: Children }) => {
   const [isSideNavExpandedState, setIsSideNavExpandedState] =
     useState(isSideNavExpanded);
 
+  // Closing the SideNav when the 'ESC' key is pressed
+  document.addEventListener('keydown', function (event) {
+    if (event.key === 'Escape') {
```

Comment:
we generally like to use our `match` helper to compare against keys:
<img width="435" alt="image" src="https://user-images.githubusercontent.com/40550942/232600024-d3ab73b4-a509-4ce7-b6f8-0ee478553151.png">
You can import it in `packages/react/src/internal/keyboard/match.js`